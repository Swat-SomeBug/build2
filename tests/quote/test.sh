#!/bin/sh

valgrind -q b -q | diff test.out -
