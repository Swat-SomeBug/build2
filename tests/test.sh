#! /usr/bin/env bash

cur_dir="`pwd`"
trap 'cd "$cur_dir"' EXIT

export PATH=$cur_dir/../build2:$PATH

function test ()
{
  echo "Testing $1"
  cd "$cur_dir/$1"
  ./test.sh
}

test "amalgam/unnamed"
test "escaping"
test "eval"
test "function/call"
test "if-else"
test "keyword"
test "names"
test "pairs"
test "quote"
test "scope"
test "variable/expansion"
test "variable/null"
test "variable/override"
test "variable/prepend"
test "variable/qualified"
test "variable/type"
