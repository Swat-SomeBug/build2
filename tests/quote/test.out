foo   bar
foo
bar
fo o
 bar 
fo o  bar 
fo o bar 
[ fo o ]
[  bar  ]
[ fo o  bar  ]
[ fo o/ bar  ]
fo obar
fo o bar 
fo obar
